### PR TITLE
feat!: Rename code scanning params from plural to singular

### PR DESCRIPTION
**Pull request by ihor-hrytskiv**
BREAKING CHANGE: Rename `RuleRequiredCodeScanningTools` to `RuleRequiredCodeScanningTool`.

Some improvements for #3256

### CODE REVIEW

File: github/repos_rules.go
Comment (gmlewis) [inline github/repos_rules.go:162]: We might as well go all in and update the comment too, agreed?
Maybe something like this?

```suggestion
// RuleRequiredCodeScanningTool represents a single required code-scanning tool for the RequiredCodeScanningParameters object.
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @ihor-hrytskiv.
One minor suggestion, then we should be ready for a second LGTM+Approval from any other contributor to this repo before merging.

**Review by gmlewis — APPROVED** [review]
Thank you, @ihor-hrytskiv !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Comment by ihor-hrytskiv** [comment]
@air-hand Could you review please?

**Review by air-hand — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @air-hand !
Merging.

**Pull request merged**
